src/space_stage/FleetMovementOrder.cs

```diff
@@ -18,22 +18,42 @@ public FleetMovementOrder(SpaceFleet fleet, Vector3 targetPosition) : base(fleet
     protected override bool WorkOnOrder(float delta)
     {
         var unitPosition = Unit.GlobalPosition;
+
         var toTarget = TargetPosition - unitPosition;
 
         var distanceToTarget = toTarget.Length();
 
         var adjustedSpeed = delta * Unit.Speed;
 
-        // TODO: adjust the fleet rotation towards the travel direction
+        // The normalized vector from the ship to the target point
+        var direction = toTarget / distanceToTarget;
+
+        var currentRotation = new Quaternion(new Vector3(0, 0, 1), Unit.Transform.Basis.Z.Normalized()).Normalized();
+        var targetRotation = new Quaternion(new Vector3(0, 0, 1), toTarget.Normalized()).Normalized();
 
-        if (distanceToTarget < adjustedSpeed)
+        if (direction.Dot(Unit.Transform.Basis.Z) < 0.99f)
         {
-            Unit.GlobalPosition = TargetPosition;
-            return true;
+            var smoothRotation = currentRotation.Slerp(targetRotation, 0.8f * delta);
+            Unit.Transform = new Transform3D(new Basis(smoothRotation), unitPosition);
+            return false;
         }
+        else
+        {
+            var smoothRotation = currentRotation.Slerp(targetRotation, 0.1f * delta);
+            Unit.Transform = new Transform3D(new Basis(smoothRotation), unitPosition);
 
-        var direction = toTarget / distanceToTarget;
-        Unit.GlobalPosition += direction * adjustedSpeed;
-        return false;
+            if (distanceToTarget < adjustedSpeed)
+            {
+                Unit.GlobalPosition = TargetPosition;
+
+                // Unit._engineEmittor.Emitting = false;
```

Comment:
I think it'd be better to make a property called `Moving` in a fleet and use that here so that the fleet class can then in the future easily be extended to handle visual changes when moving. I can do that if you need help with the code specifically.

Reply:
I would need help with that. My understanding of c# is still very limiting, and leads to many warning messages.